core/src/test/java/jenkins/plugins/DetachedPluginsUtilTest.java

```diff
@@ -27,22 +31,23 @@ public void checkJaxb() {
 
         assertEquals(new VersionNumber("11"), jaxb.getMinimumJavaVersion());
 
-        final List<DetachedPluginsUtil.DetachedPlugin> detachedPlugins = DetachedPluginsUtil.getDetachedPlugins();
+        final List<String> detachedPlugins = mapToPluginShortName(DetachedPluginsUtil.getDetachedPlugins());
         if (JavaUtils.isRunningWithJava8OrBelow()) {
-            assertEquals(0, detachedPlugins.stream()
-                    .filter(plugin -> plugin.getShortName().equals("jaxb")).count());
-        } else if (JavaUtils.getCurrentJavaRuntimeVersionNumber().isNewerThanOrEqualTo(new VersionNumber("11.0.2"))) {
-            assertEquals(1, detachedPlugins.stream()
-                    .filter(plugin -> plugin.getShortName().equals("jaxb")).count());
+            assertThat(detachedPlugins, not(hasItem("jaxb")));
+        } else {
+            assertThat(detachedPlugins, hasItem("jaxb"));
 
             final List<DetachedPluginsUtil.DetachedPlugin> detachedPluginsSince2_161 =
                     DetachedPluginsUtil.getDetachedPlugins(new VersionNumber("2.161"));
 
-            assertEquals(1, detachedPluginsSince2_161.size());
-            assertEquals("jaxb", detachedPluginsSince2_161.get(0).getShortName());
+            assertThat(mapToPluginShortName(detachedPluginsSince2_161), contains("jaxb", "trilead-api"));
```

Comment:
cosmetic test changes are so that you see the actual results when it fails, rather than 'expected: 1, was 2'

all checks here now are equivalent to before

Reply:
Thanks for the transition to clearer assertion failure messages.